core/src/main/java/org/opensearch/sql/expression/function/PPLFuncImpTable.java

```diff
@@ -432,17 +436,21 @@ public RelBuilder.AggCall resolveAgg(
     var implementation = getImplementation(functionName);
 
     // Validation is done based on original argument types to generate error from user perspective.
-    validateFunctionArgs(implementation, functionName, field, argList);
+    List<RexNode> nodes =
+        validateFunctionArgs(implementation, functionName, field, argList, context.rexBuilder);
 
     var handler = implementation.getValue();
-    return handler.apply(distinct, field, argList, context);
+    return nodes != null
+        ? handler.apply(distinct, nodes.getFirst(), nodes.subList(1, nodes.size()), context)
+        : handler.apply(distinct, field, argList, context);
   }
 
-  static void validateFunctionArgs(
+  static List<RexNode> validateFunctionArgs(
```

Comment:
To reviewer:  validateFunctionArgs return coercionNode if possible, if not, throw ExpressionEvaluationException (current behavior).